pkg/registry/apis/folders/register_test.go

```diff
@@ -114,3 +117,51 @@ func TestFolderAPIBuilder_getAuthorizerFunc(t *testing.T) {
 		})
 	}
 }
+
+func TestFolderAPIBuilder_Validate(t *testing.T) {
+	tests := []struct {
+		name  string
+		input *unstructured.Unstructured
+		err   error
+	}{
+		{
```

Comment:
Do we want to add a test case for reaching max depth as well?

Reply:
yes, but right now we can't add the test because we aren't validating it yet :)